texts/0006-tauri-releases-schedule.md

```diff
@@ -0,0 +1,53 @@
+- Feature Name: tauri_release_schedule
+- Start Date: 2024-6-6
+- RFC PR: [tauri-apps/rfcs#14](https://github.com/tauri-apps/rfcs/pull/14)
+- Tracking Issue:
+
+# Summary
+
+Define a release schedule for tauri main repo, including crates, NPM packages and CLIs. 
+
+# Motivation
+
+As it stands now, there is no release schedule and releases are made either whenever one of the working group feels like it (mainly Lucas and myself (Amr)),
+or when there is a need for an urgent fix.
+
+As a tauri user or contributor, there is no indication when a release might happen and if will it include a certain feature
+one might be waiting for which leads me to either try to figure how to use unreleased features directly from source or wait indefinitely.   
+
+A release schedule should make the releasing process more predictable for maintainers and users and prevents the tauri project from ever appearing stagnant.
+
+# Explanation
+
+We follow semver versioning for all tauri crates, NPM packages and CLIs and so we have 6 type of releases we are ever gonna make for stable releases,
+here is what each release type cycle would look like:
+
+- `patch`: weekly
+  - Can be released when needed, in the case of an urgent bug fix or security vulnerability.
+  - Releasing a `patch` version will be through cherry picking the relevant commits into a new branch and releasing it from there. 
```

Comment:
this can be problematic, if we have a feature merged in dev we'll need to cherry pick every single week - which will happen often

I'd say we compromise on releasing urgent bug fixes like we've been doing - also following Rust's pattern here

plus we don't have that many people working on tauri, so we shouldn't waste time

Reply:
I agree with Lucas, high frequency patch releases may lead to frequent updates for users, which might be disruptive.

Reply:
What about frequent releases for individual packages but less frequent for Tauri itself?